src/Haskell/Ide/Engine/Plugin/Ormolu.hs

```diff
@@ -34,24 +44,69 @@ ormoluDescriptor plId = PluginDescriptor
 
 
 provider :: FormattingProvider
-provider _contents _uri _typ _opts =
 #if __GLASGOW_HASKELL__ >= 806
-  case _typ of
-    FormatRange _ -> return $ IdeResultFail (IdeError PluginError (T.pack "Selection formatting for Ormolu is not currently supported.") Null)
-    FormatText -> pluginGetFile _contents _uri $ \file -> do
-        opts <- lookupComponentOptions file
-        let opts' = map DynOption $ filter exop $ join $ maybeToList $ componentOptions <$> opts
-            conf  = Config opts' False False True False
-        result <- liftIO $ try @OrmoluException (ormolu conf file (T.unpack _contents))
-
-        case result of
-          Left  err -> return $ IdeResultFail (IdeError PluginError (T.pack $ "ormoluCmd: " ++ show err) Null)
-          Right new -> return $ IdeResultOk [TextEdit (fullRange _contents) new]
-  where
-    exop s =
-      "-X" `isPrefixOf` s
-      || "-fplugin=" `isPrefixOf` s
-      || "-pgmF=" `isPrefixOf` s
+{-# LANGUAGE BlockArguments #-}
```

Comment:
Do we really need this extension? It just adds cognitive overload, people reading the code have to go off and learn what it does before continuing.

Reply:
Oops we don't even need it anymore